Add documentation related to the rasa helm chart (#9077)
RasaHQ/rasa: docs/docs/how-to-deploy.mdx

* Custom: Read the above, as well as the [Advanced Configuration](https://rasa.com/docs/rasa-x/installation-and-setup/customize/#helm-chart)
  documentation, and customize the [open source Helm charts](https://github.com/RasaHQ/rasa-x-helm) to your needs.

## Deploying a Rasa Open Source Assistant

While the above deployment methods involve deploying an assistant with Rasa X, the following instructions describe how to deploy a Rasa Open Source server only

> by using [the rasa helm chart](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa) in a scalable cluster environment using OpenShift or Kubernetes (K8S).

Reviewer: ```suggestion
by using the [Rasa Helm Chart](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa) in a scalable cluster environment using OpenShift or Kubernetes (K8S).
```

### Cluster Requirements

To install the Rasa Helm chart, you need an existing
[Kubernetes cluster](https://kubernetes.io/) or [OpenShift cluster](https://www.openshift.com/).
Setting up a Kubernetes / OpenShift cluster can be tedious, hence we
recommend to get a managed cluster from a cloud provider like
[Google Cloud](https://cloud.google.com/kubernetes-engine),
[DigitalOcean](https://www.digitalocean.com/products/kubernetes/),
[Microsoft Azure](https://azure.microsoft.com/en-us/services/kubernetes-service/), or
[Amazon EKS](https://aws.amazon.com/eks/).

:::note

> The Rasa Helm chart is open-source and available in the

Reviewer: ```suggestion
The Rasa Helm chart is open source and available in the
```

[helm-charts repository](https://github.com/rasahq/helm-charts).
Please
[create an issue](https://github.com/RasaHQ/helm-charts/issues/new) in this
repository if you discover bugs or have suggestions for improvements.

:::

### Installation Requirements

> 1. Please check that you installed the Kubernetes or OpenShift command line

Reviewer 2: ```suggestion
1. Check that you have installed the Kubernetes or OpenShift command line
```

   [OpenShift CLI](https://docs.openshift.com/container-platform/4.7/cli_reference/openshift_cli/getting-started-cli.html#installing-openshift-cli)
   depending on the cluster you’re using.

2. Make sure that the Kubernetes / OpenShift CLI is correctly connected to
   your cluster. You can do so by using the following commands:

   <Tabs values={[{"label": "Kubernetes", "value": "kubernetes"}, {"label": "OpenShift", "value": "openshift"}]} defaultValue="kubernetes">
     <TabItem value="kubernetes">

     ```bash
     kubectl version --short

     # The output should be similar to this
     # Client Version: v1.19.11
     # Server Version: v1.19.10
     ```

     </TabItem>
     <TabItem value="openshift">

     ```bash
     oc version

     # The output should be similar to this
     # Client Version: 4.7.13
     # Kubernetes Version: v1.20.0+df9c838
     ```

     </TabItem>
   </Tabs>

   If you get an error when executing the command, you are not connected to your
   cluster. To get the command to connect to the cluster please consult your cluster’s
   admin or the documentation of your cloud provider.

> 3. Please make sure you have the [Helm CLI](https://helm.sh/docs/intro/install/)

Reviewer 2: ```suggestion
3. Make sure you have the [Helm CLI](https://helm.sh/docs/intro/install/)
```

   installed. To check this, run:

   ```bash
   helm version --short

   # The output should be similar to this
   # v3.6.0+g7f2df64
   ```

   If this command leads to an error, please install the
   [Helm CLI](https://helm.sh/docs/intro/install/).

   In case you are using a version `<3.5` of Helm, please update to Helm version
   `>=3.5`.

### Installation

#### 1. Create Namespace

> We recommend installing Rasa Assistant in a separate

Reviewer: ```suggestion
We recommend installing the Rasa Open Source server in a separate
```

Reviewer: Or maybe just "Rasa Open Source" (without server?)

Author: "Rasa Open Source" sounds good to me 😄

[namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)
to avoid interfering with existing cluster deployments. To create a new namespace
run the following command:

<Tabs values={[{"label": "Kubernetes", "value": "kubernetes"}, {"label": "OpenShift", "value": "openshift"}]} defaultValue="kubernetes">
  <TabItem value="kubernetes">

  ```bash
  kubectl create namespace <your namespace>
  ```

  </TabItem>
  <TabItem value="openshift">

  ```bash
  oc create namespace <your namespace>
  ```

  </TabItem>
</Tabs>

#### 2. Create Values File

Prepare an empty file called `rasa-values.yml` which will include all your custom
configuration for the installation with Helm.

> All available values you can find in [the rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa#values).
> 
> :::note
> The default configuration of the Rasa chart deploys a Rasa Open Source Server, downloads a model, and serves the downloaded model.
> Visit [the rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa#quick-start) to check out more examples of configuration.

Reviewer 2: Let's decide consistently if we wnat to capitalize `rasa` in this
```suggestion
All available values you can find in [the Rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa#values).

:::note
The default configuration of the Rasa chart deploys a Rasa Open Source Server, downloads a model, and serves the downloaded model.
Visit [the Rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa#quick-start) to check out more examples of configuration.
```

:::

#### 3. Loading an initial model

The first time you install Rasa, you may not have a model server available yet, or you may want an lightweight model for testing the deployment.
For this purpose, you can choose between training or downloading an initial model. By default, the Rasa chart downloads an example model from GitHub.
To use this option, you don't have to change anything.

If you want to define an existing model to download from a URL you define instead, update your `rasa-values.yaml` with the URL according to the following configuration:

  ```yaml
  applicationSettings:
    initialModel: "https://github.com/RasaHQ/rasa-x-demo/blob/master/models/model.tar.gz?raw=true"
  ```
:::note
The URL for the initial model download has to point to a tar.gz file and must not require authentication.

:::

If you want to train an initial model you can do this by setting the `applicationSettings.trainInitialModel` to `true`.
It creates a init container that trains a model based on data located in the `/app` directory. If the `/app` directory is empty it creates a new project.

> [Here](https://github.com/RasaHQ/helm-charts/blob/main/examples/rasa/train-model-helmfile.yaml) you can find an example that shows how to download data files from a git repository and train an initial model.

Reviewer 2: ```suggestion
You can find an example that shows how to download data files from a git repository and train an initial model in the Rasa helm charts [examples](https://github.com/RasaHQ/helm-charts/blob/main/examples/rasa/train-model-helmfile.yaml).
```

By default the Rasa deployment is exposed via the `rasa` (`<release name>`) service and accessible only within a Kubernetes cluster. You can get
the IP address using this command:

<Tabs values={[{"label": "Kubernetes", "value": "kubernetes"}, {"label": "OpenShift", "value": "openshift"}]} defaultValue="kubernetes">
  <TabItem value="kubernetes">

  ```bash
    export SERVICE_PORT=$(kubectl get --namespace <your namespace> -o jsonpath="{.spec.ports[0].port}" services <release name>)
    kubectl port-forward --namespace <your namespace> svc/<release name> ${SERVICE_PORT}:${SERVICE_PORT} &
  ```

  </TabItem>
  <TabItem value="openshift">

  ```bash
    export SERVICE_PORT=$(oc get --namespace <your namespace> -o jsonpath="{.spec.ports[0].port}" services <release name>)
    oc port-forward --namespace <your namespace> svc/<release name> ${SERVICE_PORT}:${SERVICE_PORT} &
  ```

  </TabItem>
</Tabs>

You can then access the deployment on `http://127.0.0.1:${SERVICE_PORT}`

Visit [the Rasa helm chart README](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa#exposing-the-rasa-deployment-to-the-public) to learn other ways to expose your deployment.

#### Next Steps

> Visit [the Rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa) where you can find examples of configuration and learn how to e.g. integrate Rasa Open Source deployment with Rasa X.

Reviewer 2: ```suggestion
Visit [the Rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa) where you can find examples of configuration and learn how to e.g. integrate your Rasa Open Source deployment with Rasa X.
```
